CourseGrab/LoginViewController.swift

```diff
@@ -2,21 +2,99 @@
+class LoginViewController: UIViewController, GIDSignInDelegate {
+
+    private let stackView = UIStackView()
+    private let titleLabel = UILabel()
+    private let logoView = UIImageView()
+    private let signInButton = GIDSignInButton()
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
 
         view.backgroundColor = .white
+
+        GIDSignIn.sharedInstance().delegate = self
+        GIDSignIn.sharedInstance()?.presentingViewController = self
+        // Automatically sign in the user.
+        GIDSignIn.sharedInstance()?.restorePreviousSignIn()
+
+        setUpStackView()
+        setUpTitleLabel()
+        setUpLogoView()
+        setUpSignInButton()
+
+    }
+
+    private func setUpStackView() {
+        stackView.axis = .vertical
+        stackView.distribution = .fill
+        stackView.alignment = .center
+        stackView.spacing = 40
+        view.addSubview(stackView)
+
+        stackView.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview().inset(32)
+            make.centerY.equalToSuperview().priority(.high)
+        }
+    }
+
+    private func setUpTitleLabel() {
+        titleLabel.text = "CourseGrab"
+        titleLabel.textColor = .black
+        titleLabel.textAlignment = .center
+        titleLabel.font = .systemFont(ofSize: 36, weight: .bold)
+        stackView.addArrangedSubview(titleLabel)
     }
 
+    private func setUpLogoView() {
+        logoView.image = UIImage(named: "image")
+        logoView.contentMode = .scaleAspectFit
+        stackView.addArrangedSubview(logoView)
+
+        logoView.snp.makeConstraints { make in
+            make.height.equalTo(128)
+        }
+    }
+
+    private func setUpSignInButton() {
+        stackView.addArrangedSubview(signInButton)
+    }
+
+    func sign(_ signIn: GIDSignIn!, didSignInFor user: GIDGoogleUser!, withError error: Error!) {
+        if let error = error {
+            if (error as NSError).code == GIDSignInErrorCode.hasNoAuthInKeychain.rawValue {
+                // User is not signed in or have signed out
+            } else {
+                //print("\(error.localizedDescription)")
+            }
+            return
+        }
+        let coursesViewController = CoursesViewController()
+        let coursesNavigationController = CoursesNavigationController(rootViewController: coursesViewController)
+        view.window?.rootViewController = coursesNavigationController
+        // We can remove this but this is what was in the documentation
+        // Perform any operations on signed in user here.
+//        let userId = user.userID // For client-side use only!
+//        let idToken = user.authentication.idToken // Safe to send to the server
+//        let fullName = user.profile.name
+//        let givenName = user.profile.givenName
+//        let familyName = user.profile.familyName
+//        let email = user.profile.email
```

Comment:
Yeah I would remove. For future reference, could we use `user.authentication.idToken` instead of the email to identify users? Is there a point? We can discuss this later. I would remove the comments either way.